README.md

```diff
@@ -2,40 +2,72 @@ Simple Serverless Keyed Event Log
-- a script that lets you download the data (and that you can use a starting point for more complicated data processing)
+This is the code for a [Firebase](https://firebase.google.com) project that
+provides an integrated solution to this problem. Specifically, it defines:
+
+- an [HTTP-triggered
+  function](https://firebase.google.com/docs/functions/http-events) providing
+  endpoint you can query with your data
+- the [database](https://firebase.google.com/docs/firestore/) (i.e., the
+  configuration for it) where the data gets stored
+- a [hosted](https://firebase.google.com/docs/hosting) webpage that allows quick
+  access to the data
+- a script that lets you download the data (and that you can use a starting
+  point for more complicated data processing)
 
 With Firebase, you have the option of the function running under a custom domain name that you bring in.
 
-
 ## Usage
 ### Storing data
-To store data, make a POST request to `https://your_project_url/save` with the following fields:
-
-- `value`, the data you want to store
-- `key`, the key you want to use to identify this data (it need not be unique). For example, this could be the type of the event you're logging.
-- `uid`, an identifier for the user associated with this event
-- `client`, an identifier for the client issuing this request (e.g., the app version)
-
-These fields are required, but the data types are not enforced. Especially useful for `value`, they can be nested JSON objects.
-
-#### No authentication
-**Anyone can write to the database.** Requests are neither authenticated nor validated. It is assumed that you have an ex post facto way of identifying valid requests.
-
-
-## Previewing data
-To see your data on a webpage, go to `https://your_project_url/get`. You will need to authenticate with a whitelisted email (see Setup below).
-
-
-## Downloading data
+To store data, make a POST request to `https://your_project_url/save` with the
+following **required** fields:
+
+- `key` (string|any), the key you want to use to identify this data (it need not be unique).
+  For example, this could be the type of the event you're logging.
+- `value` (any), the data you want to store. Can be any type, including nested
+  JSON objects.
+- `uid` (string|any), an identifier for the user associated with this event
+- `clientVersion` (string|any), an identifier for the client issuing this request (e.g., the
+  app version). Think of this as the client user-agent.
+- `timestamp` (string|any), the timestamp of when the client generated the event. NOTE: This
+  should be when the event actually occurred on the client and NOT when it was
+  sent to the server.
+
+Example event:
+```javascript
+{
+  key: "login_event",
+  value: {
+    something: "abc",
+    other: "xyz",
+    another: {
+      thing: "def"
+    }
+  },
+  uid: "123",
+  clientVersion: "v0.1.2",
+  timestamp: "2020-04-22T19:09:04.641Z",
+  customFieldA: "you can add as many custom fields as you want at the top level",
```

Comment:
I'd prefer custom fields to be down in `value`. That way, the top level has a finite and final list of fields, and we know that any additional or metadata fields we might add are out of the client's control.